mm2src/coins/lightning.rs

```diff
@@ -697,26 +670,97 @@ impl SwapOps for LightningCoin {
     fn check_if_my_payment_sent(
         &self,
-        _if_my_payment_spent_args: CheckIfMyPaymentSentArgs<'_>,
+        if_my_payment_sent_args: CheckIfMyPaymentSentArgs<'_>,
     ) -> Box<dyn Future<Item = Option<TransactionEnum>, Error = String> + Send> {
-        Box::new(futures01::future::ok(None))
+        let PaymentInstructions::Lightning(invoice) = try_f!(if_my_payment_sent_args
+            .payment_instructions
+            .clone()
+            .ok_or("payment_instructions can't be None"));
+        let payment_hash = PaymentHash((invoice.payment_hash()).into_inner());
+        let payment_hex = hex::encode(payment_hash.0);
+        let coin = self.clone();
+        let fut = async move {
+            match coin.db.get_payment_from_db(payment_hash).await {
+                Ok(maybe_payment) => Ok(maybe_payment.map(|p| p.payment_hash.into())),
+                Err(e) => ERR!(
+                    "Unable to check if payment {} is in db or not error: {}",
+                    payment_hex,
+                    e
+                ),
+            }
+        };
+        Box::new(fut.boxed().compat())
     }
 
+    // Todo: need to also check on-chain spending
     async fn search_for_swap_tx_spend_my(
         &self,
-        _: SearchForSwapTxSpendInput<'_>,
+        input: SearchForSwapTxSpendInput<'_>,
     ) -> Result<Option<FoundSwapTxSpend>, String> {
-        unimplemented!()
+        let payment_hash = payment_hash_from_slice(input.tx).map_err(|e| e.to_string())?;
+        let payment_hex = hex::encode(payment_hash.0);
+        match self.db.get_payment_from_db(payment_hash).await {
+            Ok(Some(payment)) => {
+                if !payment.is_outbound() {
+                    return ERR!("Payment {} should be an outbound payment!", payment_hex);
+                }
+                match payment.status {
+                    HTLCStatus::Pending => Ok(None),
+                    HTLCStatus::Succeeded => Ok(Some(FoundSwapTxSpend::Spent(TransactionEnum::LightningPayment(
+                        payment_hash,
+                    )))),
+                    HTLCStatus::Received => {
+                        ERR!(
+                            "Payment {} has an invalid status of {} in the db",
+                            payment_hex,
+                            payment.status
+                        )
+                    },
+                    HTLCStatus::Failed => Ok(Some(FoundSwapTxSpend::Refunded(TransactionEnum::LightningPayment(
+                        payment_hash,
+                    )))),
+                }
+            },
+            Ok(None) => ERR!("Payment {} is not in the database when it should be!", payment_hex),
+            Err(e) => ERR!(
+                "Unable to retrieve payment {} from the database error: {}",
+                payment_hex,
+                e
+            ),
+        }
     }
 
+    // Todo: need to also check on-chain spending
```

Comment:
Do you plan to do it on next iteration? 🙂

Reply:
Yes, this should be done in next sprint. This can only happen when the channel the payment is sent/received over is closed before the swap finishes, I believe it might be easier to implement this if `Event::ChannelClosed` from rust-lightning side provides info about unsettled HTLCs in the channel closing transaction.https://github.com/KomodoPlatform/atomicDEX-API/blob/ae5daf7c082b3493555057a7c7559e6d5d211893/mm2src/coins/lightning/ln_events.rs#L85 But if this can't be done easy from rust-lightning side I believe it can be done on mm2 straight away. Just need to check the rust-lightning code in details for unsettled HTLCs and make sure I don't miss anything before implementing this, that's why I moved it to next sprint.